app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  before_action :authorize_user
+
+  def show
+    @user = User.new
+    @users = User.order(:username)
+
+    if params[:user] != nil
+      username = params[:user][:username]
+      @user = User.find_by(username: username)
+    end
+  end
+
+  def index
+    if params[:user][:id]
+      @selected_user = User.find(params[:user][:id])
+    else
+      @selected_user = User.find(params[:id])
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
```

Comment:
if we override the `to_param` like I mentioned above, we'll have to do a `find_by(username: params[:id])` here